pkg/common/cns-lib/vsphere/utils.go

```diff
@@ -413,16 +409,30 @@ func GetCandidateDatastoresInCluster(ctx context.Context, vc *VirtualCenter, clu
 	if len(hosts) == 0 {
 		return nil, nil, fmt.Errorf("empty List of hosts returned from VC")
 	}
+
 	sharedDatastores := make([]*DatastoreInfo, 0)
 	vsanDirectDatastores := make([]*DatastoreInfo, 0)
 	for index, host := range hosts {
 		accessibleDatastores, err := host.GetAllAccessibleDatastores(ctx)
 		if err != nil {
 			return nil, nil, err
 		}
+		if len(accessibleDatastores) == 0 {
```

Comment:
can there be a case where in single vsan/vc cluster, there are some hosts without any disks/datastore associated and some hosts having vsan direct datastores ?? if yes, do we need to continue here even if host does not have any datastore accessible?

Reply:
If you are only looking for vSAN Direct datastores in the cluster and one or more hosts do not have any accessible datastores, then yes we should not error out here. Let me check if we need to handle this scenario.

Reply:
Removed the code after having a conversation with Divyen